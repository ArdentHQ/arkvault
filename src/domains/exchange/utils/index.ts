export * from "./assertions";
export * from "./errors";
