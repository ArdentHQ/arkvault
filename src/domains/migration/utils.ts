import { MigrationTransactionStatus } from "@/domains/migration/migration.contracts";

export const getIcon = (status: MigrationTransactionStatus) => {
	if (status === MigrationTransactionStatus.Confirmed) {
		return {
			color: "text-theme-success-600",
			name: "CircleCheckMark",
		};
	}

	return {
		color: "text-theme-warning-300",
		name: "Clock",
	};
};
