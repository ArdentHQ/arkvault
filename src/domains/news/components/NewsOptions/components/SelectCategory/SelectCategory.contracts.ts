export type SelectCategoryProperties = {
	children: React.ReactNode;
	type?: "radio" | "checkbox";
	name?: string | number;
	value?: string | number;
	checked?: boolean;
	defaultChecked?: boolean;
	disabled?: boolean;
} & React.HTMLProps<any>;
