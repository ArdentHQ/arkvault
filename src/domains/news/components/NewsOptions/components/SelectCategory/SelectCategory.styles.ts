import tw, { styled } from "twin.macro";

export const Input = styled("input", { target: "select-category-input" })`
	${tw`sr-only`}
`;

export const CustomButton = styled.div`
	${tw`p-2 px-4 font-semibold text-center transition-colors duration-200 border-2 rounded-md border-theme-primary-100 text-theme-secondary-700 hover:border-theme-primary-400 cursor-pointer dark:text-theme-secondary-200 dark:border-theme-secondary-800 dark:hover:border-theme-primary-400`}
	${Input}:checked + & {
		${tw`border-theme-primary-600 bg-theme-primary-50 dark:bg-theme-primary-900 dark:border-theme-primary-600 `}
	}
`;
