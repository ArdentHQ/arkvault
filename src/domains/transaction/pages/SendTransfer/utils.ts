export const getFeeType = (numberOfRecipients: number): "multiPayment" | "transfer" =>
	numberOfRecipients > 1 ? "multiPayment" : "transfer";
