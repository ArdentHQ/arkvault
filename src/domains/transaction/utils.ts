import { Services } from "@payvo/sdk";

export const isNoDeviceError = (error: any) => {
	if (!error) {
		return false;
	}

	if (String(error).includes("no device found")) {
		return true;
	}

	return false;
};

export const isRejectionError = (error: any) => String(error).includes("Condition of use not satisfied");

export const handleBroadcastError = ({ errors }: Services.BroadcastResponse) => {
	const allErrors = Object.values(errors);

	if (allErrors.length === 0) {
		return;
	}

	throw new Error(allErrors[0]);
};

export const withAbortPromise =
	(signal?: AbortSignal, callback?: () => void) =>
	<T>(promise: Promise<T>) =>
		new Promise<T>((resolve, reject) => {
			if (signal) {
				signal.addEventListener("abort", () => {
					callback?.();
					reject("ERR_ABORT");
				});
			}

			return promise.then(resolve).catch(reject);
		});
