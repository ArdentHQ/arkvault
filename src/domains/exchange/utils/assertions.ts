import { AssertionError } from "assert";
import { Contracts } from "@ardenthq/sdk-profiles";

import { CurrencyData } from "@/domains/exchange/exchange.contracts";
import { ExchangeService } from "@/domains/exchange/services/exchange.service";

export function assertExchangeService(exchangeService?: ExchangeService): asserts exchangeService is ExchangeService {
	if (!(exchangeService instanceof ExchangeService)) {
		throw new AssertionError({
			message: `Expected 'exchangeService' to be ExchangeService, but received ${exchangeService}`,
		});
	}
}

export function assertExchangeTransaction(
	exchangeTransaction?: Contracts.IExchangeTransaction,
): asserts exchangeTransaction is Contracts.IExchangeTransaction {
	if (typeof exchangeTransaction?.orderId !== "function") {
		throw new AssertionError({
			message: `Expected 'exchangeTransaction' to be Contracts.IExchangeTransaction, but received ${exchangeTransaction}`,
		});
	}
}

export function assertCurrency(currencyData?: CurrencyData): asserts currencyData is CurrencyData {
	if (!currencyData || typeof currencyData !== "object" || !("coin" in currencyData && "name" in currencyData)) {
		throw new AssertionError({
			message: `Expected 'currencyData' to be CurrencyData, but received ${currencyData}`,
		});
	}
}
