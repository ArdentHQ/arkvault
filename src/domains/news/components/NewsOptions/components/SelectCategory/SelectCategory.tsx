import React from "react";

import { SelectCategoryProperties } from "./SelectCategory.contracts";
import { Input, CustomButton } from "./SelectCategory.styles";

export const SelectCategory = React.forwardRef<HTMLInputElement, SelectCategoryProperties>(
	(
		{
			children,
			type = "checkbox",
			name,
			value,
			checked,
			defaultChecked,
			disabled,
			onChange,
			...properties
		}: SelectCategoryProperties,
		reference,
	) => (
		<label htmlFor={name} className="cursor-pointer" {...properties}>
			<Input
				data-testid={name ? `SelectCategory__${name}` : undefined}
				ref={reference}
				type={type}
				name={name}
				value={value}
				checked={checked}
				defaultChecked={defaultChecked}
				disabled={disabled}
				onChange={onChange}
			/>
			<CustomButton>{children}</CustomButton>
		</label>
	),
);

SelectCategory.displayName = "SelectCategory";
