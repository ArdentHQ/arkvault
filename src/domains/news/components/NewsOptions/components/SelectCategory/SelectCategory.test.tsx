import React from "react";

import { SelectCategory } from "./SelectCategory";
import { render } from "@/utils/testing-library";

describe("SelectCategory", () => {
	it("should render", () => {
		const { container, asFragment } = render(<SelectCategory name="category">#All</SelectCategory>);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});
});
