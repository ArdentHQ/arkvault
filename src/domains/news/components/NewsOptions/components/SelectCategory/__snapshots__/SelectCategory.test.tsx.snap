// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectCategory should render 1`] = `
<DocumentFragment>
  <label
    class="cursor-pointer"
    for="category"
  >
    <input
      class="css-1r71f0z select-category-input"
      data-testid="SelectCategory__category"
      name="category"
      type="checkbox"
      value=""
    />
    <div
      class="css-fn0s68"
    >
      #All
    </div>
  </label>
</DocumentFragment>
`;
